### PR TITLE
feat(cli): refactor and add album support

### PR DESCRIPTION
**Pull request by etnoy**
* Deploy CLI as npm package @immich/cli
* Remove read-only asset support
* Add support for automatic album creation
* Pull supported file extensions from server
* Improved documentation
* Use SDK instead of manual calls
* Make server-info command richer

How this has been tested:
* Uploading various folders to immich
* Verifying created albums are as expected

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -2,21 +2,27 @@ A command-line interface for interfacing with Immich
...
 
-    $ ts-node cli/src login-key https://your-immich-instance/api your-api-key
+    $ immich login-key https://your-immich-instance/api your-api-key
 
 NOTE: This will store your api key under ~/.config/immich/auth.yml
 
 Next, you can run commands:
 
-    $ ts-node cli/src server-info
+    $ immich cli/src server-info
```
Comment (jrasm91) [inline cli/README.md:21]: No need to pass src/* anymore

Comment (etnoy) [reply cli/README.md:21]: Can't believe I missed that one!

**Comment by alextran1502** [comment]
Hello @etnoy, thank you for your continuous work on this. I just tested the PR and saw an issue about album creation.

Here is what the file systems look like

image

In `Cute` and `With Da Space` there are more assets.

Here is the command I used

image

When they get uploaded to the server, here is the result in the Albums tab

image

And here is one of the album's name

image

So I guess the album was created based on the absolute path of the file.

I expected there will be two album that get created namely `Cute` and `With Da Space` while the rest that are not in any folder would not be put into any album

**Comment by etnoy** [comment]
> Hello @etnoy, thank you for your continuous work on this. I just tested the PR and saw an issue about album creation.
> 
> Here is what the file systems look like
> 
> image
> 
> In `Cute` and `With Da Space` there are more assets.
> 
> Here is the command I used
> 
> image
> 
> When they get uploaded to the server, here is the result in the Albums tab
> 
> image
> 
> And here is one of the album's name
> 
> image
> 
> So I guess the album was created based on the absolute path of the file.
> 
> I expected there will be two album that get created namely `Cute` and `With Da Space` while the rest that are not in any folder would not be put into any album

Thanks for testing. This might be a Windows issue, it works on Linux. Also, I copied the logic from the legacy CLI so the same issue is likely they're too. Will look into this later today

**Comment by etnoy** [comment]
> Hello @etnoy, thank you for your continuous work on this. I just tested the PR and saw an issue about album creation.
> 
> Here is what the file systems look like
> 
> image
> 
> In `Cute` and `With Da Space` there are more assets.
> 
> Here is the command I used
> 
> image
> 
> When they get uploaded to the server, here is the result in the Albums tab
> 
> image
> 
> And here is one of the album's name
> 
> image
> 
> So I guess the album was created based on the absolute path of the file.
> 
> I expected there will be two album that get created namely `Cute` and `With Da Space` while the rest that are not in any folder would not be put into any album

I've added a check for windows, try again :)

**Comment by alextran1502** [comment]
Looks good, can you help fix the documentation build error?

image

Then feel free to merge

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**